apps/studio/prisma/migrations/20240923090529_add_resource_perms/migration.sql

```diff
@@ -0,0 +1,18 @@
+-- CreateTable
+CREATE TABLE "ResourcePermission" (
+    "userId" TEXT NOT NULL,
+    "siteId" INTEGER NOT NULL,
+    "resourceId" BIGINT NOT NULL,
+    "role" "RoleType" NOT NULL,
+
+    CONSTRAINT "ResourcePermission_pkey" PRIMARY KEY ("siteId","userId","resourceId")
+);
+
```

Comment:
issue: the trigger for the updatedAt needs to be manually added (once you add the field in the prisma schema)

```suggestion

-- AlterTable
CREATE TRIGGER update_timestamp BEFORE UPDATE ON "ResourcePermission" FOR EACH ROW EXECUTE PROCEDURE moddatetime("updatedAt");

```

Reply:
oh this is abit interesting - why do we need the trigger ah? went to peek at the prisma [docs](https://www.prisma.io/docs/orm/reference/prisma-schema-reference#updatedat) and they said that it's automatically added! 

let me try running this to the schema and check if it's updated

Reply:
eh cool, it doesn't work without the trigger. i tried with kysely and manual updates but `updatedAt` doesn't work. going to do the `trigger`

Reply:
It's cos we are using Kysely as the query builder and only using Prisma for the models and db client. Usually on Prisma with the `@updatedAt` decorator, Prisma (as the ORM) will automatically perform the update on the updatedAt field, but we are not using Prisma in our queries here so we have to do it ourselves.

Can see #609 for more context.